android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt

```diff
@@ -776,7 +908,7 @@ open class Ampli {
 
             try {
                 value?.let {
-                    json.put(key, if (value is Collection<*>) JSONArray(value) else value)
+                    json.put(key, if (isArray(value)) getJsonArray(value) else value)
```

Comment:
@falconandy can you please explain this change? Does `is Collection<*>` not work for certain cases?

Reply:
`quicktype` generates `List` type - `is Collection<*>` check works. Now we returned to `Array` type and `is Collection<*>` check doesn't work